validator/enforce_labels_test.rego

```diff
@@ -86,122 +88,185 @@ compute_snapshot_violations[violation] {
 	violation := violations[_]
 }
 
+bigtable_violations[violation] {
+	constraints := [fixture_constraints.require_labels]
+	violations := find_all_violations with data.resources as fixture_bigtable
+		 with data.test_constraints as constraints
+
+	violation := violations[_]
+}
+
+cloudsql_violations[violation] {
+	constraints := [fixture_constraints.require_labels]
+	violations := find_all_violations with data.resources as fixture_cloudsql
+		 with data.test_constraints as constraints
+
+	violation := violations[_]
+}
+
 ##### Testing for projects
 
-# Confirm four violations were found for all projects
-# 3 projects have violations - 1 of which has 2 violations (missing 2 labels) 
-test_enforce_label_projects_violates_four {
+# Confirm six violations were found for all projects
+# 4 projects have violations - 2 of which have 2 violations (one has 2 labels missing, the other has 2 labels with invalid values) 
+test_enforce_label_projects_violates_six {
 	violations := project_violations
-	count(violations) == 4
+	count(violations) == 6
 	violation := violations[_]
 	is_string(violation.msg)
 	is_object(violation.details)
 }
 
-# confirm which 3 projects are in violation
+# confirm which 4 projects are in violation
```

Comment:
can we also verify that there are no other projects in violation (similarly below)? otherwise we're only checking for subset, right?

example: https://github.com/forseti-security/policy-library/blob/master/validator/gke_allowed_node_sa_scopes_test.rego#L51

Reply:
Great thinking. I updated all the tests to reflect this change and I merged the count and comparison ones into a single test, so that it's easier to read and maintain.